web/src/pages/loading.tsx

```diff
@@ -37,18 +42,14 @@ const LoadingPage = (): ReactElement => {
         encounteredMyNjLinkingError: true,
       });
     } else {
-      if (loginPageEnabled) {
-        router && router.push(ROUTES.login);
-      } else {
-        triggerSignIn();
-      }
+      triggerSignIn();
```

Comment:
tolerable but non-ideal edge case:
- an authenticated user navigates to `/loading`
- the other `if` conditions are false, so `triggerSignIn` is called
- user gets redirected from myNJ with a new `code` query parameter
- user ends up on dashboard

it's unlikely someone will just navigate to `/loading` by themselves, and it at least lands them in the right spot, but I don't love it. Adding this should avoid it:
```ts
getActiveUser().then(() => {
// The dispatch call should have already happened on
// on line 32 in a previous render, so do nothing and defer
// any routing logic to `useMountEffectWhenDefined` below
}.catch(() => {
    triggerSignIn();
})
```
but maybe that's overkill for an unlikely scenario?

Reply:
Never mind. I reverted this logic to what it was before. If the login page is enabled, then the user will get directed to login and avoid the additional `triggerSignIn` call. The small amount of cases left in the `else` case is negligible for now.